### PR TITLE
feat: add tencent asr

### LINKED ISSUE #6090: Feat: add support for tencent ASR
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

We want fast, low-cost, robust ASR services in mainland China and Tencent Cloud provides such service.
We already integrated it with dify, will submit PR for review

### 2. Additional context or comments

_No response_

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by lancemao**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Tencent ASR provides low cost, fast, robust speech to text services, we'd like user to have such option

Close https://github.com/langgenius/dify/issues/6090

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A. try to add Tencent ASR with incorrect credential, it should show error message returned by tencent
- [x] Test B. after providing correct credential, add speech to your agent and ASR should work as expected (including hot words which can be config in tencent's console)

### CODE REVIEW

File: api/core/model_runtime/model_providers/_position.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@
 - tongyi
 - wenxin
 - moonshot
+- tencent
```
Comment (takatost) [inline api/core/model_runtime/model_providers/_position.yaml:26]: tencent_cloud might be better?

File: api/core/model_runtime/model_providers/tencent/speech2text/flash_recognizer.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import requests
+
+
+class Credential:
+    def __init__(self, secret_id, secret_key):
+        self.secret_id = secret_id
+        self.secret_key = secret_key
+
+
+# 录音识别极速版使用
```
Comment (takatost) [inline api/core/model_runtime/model_providers/tencent/speech2text/flash_recognizer.py]: To facilitate community co-creation, please change the comment to English, thank you!

File: api/core/model_runtime/model_providers/tencent/speech2text/speech2text.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        Map model invoke error to unified error
+        The key is the error type thrown to the caller
+        The value is the error type thrown by the model,
+        which needs to be converted into a unified error type for the caller.
+
+        :return: Invoke error mapping
+        """
+        return {
+            InvokeConnectionError: [
+                InvokeConnectionError
```
Comment (takatost) [inline api/core/model_runtime/model_providers/tencent/speech2text/speech2text.py]: The error here refers to the exception returned by tencent cloud, which is converted to an error by mapping here.

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/tencent/speech2text/flash_recognizer.py (hunk 2 above)
Comment (lancemao) [reply api/core/model_runtime/model_providers/tencent/speech2text/flash_recognizer.py]: flash_recognizer is copied from tencent SDK

File: api/core/model_runtime/model_providers/_position.yaml (hunk 1 above)
Comment (lancemao) [reply api/core/model_runtime/model_providers/_position.yaml:26]: other module like tencent cos also use 'tencent', I think it's fine

File: api/core/model_runtime/model_providers/tencent/speech2text/flash_recognizer.py (hunk 2 above)
Comment (lancemao) [reply api/core/model_runtime/model_providers/tencent/speech2text/flash_recognizer.py]: fixed

File: api/core/model_runtime/model_providers/tencent/speech2text/speech2text.py (hunk 3 above)
Comment (lancemao) [reply api/core/model_runtime/model_providers/tencent/speech2text/speech2text.py]: seems that tencent not returning error. Added requests.exceptions.ConnectionError

**Review by crazywoola — APPROVED** [review]

**Review by takatost — APPROVED** [review]

**Pull request merged**